test/e2e/cypress/support/commands.js

```diff
@@ -85,6 +85,25 @@ Cypress.Commands.add('updateApiKeyExpiration', (apiKeyExpiration) => {
   });
 });
 
+Cypress.Commands.add('preloadTestData', () => {
+  cy.isTestDataLoaded().then((isLoaded) => {
+    if (!isLoaded) cy.loadScenario('healthy-27-node-SAP-cluster');
+  });
+  cy.loadScenario('healthy-27-node-SAP-cluster');
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** ‏Data is loaded regardless of the result of `isTestDataLoaded`. Also, in the case of `isLoaded==false`, it's loaded twice

Reply:
in some tests it is needed to be loaded twice, so for some tests it will be loaded once and for some others it will be loaded twice. This is required for various reasons
1. Add ability to run single spec files
2. Restore changes that previous tests have done
3. To fully load the test scenario we need to load scenario twice as some data require some previous [pre conditions](https://suse.slack.com/archives/C02U4TWGKSQ/p1734447399649439?thread_ts=1734418842.256039&cid=C02U4TWGKSQ) in order to be loaded in trento.

Reply:
Your point is good, I'm just afraid the actual code does not respect what you want to achieve.

As far as I understand, every time `preloadTestData` is executed it does the following:
* it always loads the scenario once (line 92)
* it checks data has been loaded (line 90)
* if not, it loads the scenario again (line 90)

Thus data is olways loaded once per spec file, and `isLoaded` will always be true. Isn't it?

Don't we just need to remove line 92?
```suggestion
  cy.isTestDataLoaded().then((isLoaded) => {
    if (!isLoaded) cy.loadScenario('healthy-27-node-SAP-cluster');
  });
```

Reply:
when running the full test suite (or a subset of tests), it will only be false the first time, that's for sure. With this logic what we avoid is to call loadScenario twice when it's not needed and to have a consistent test scenario, as there are things that in the first shot of photofinish, trento doesn't load but it does in the second time.

Reply:
Given `preloadTestData` is called in 13 specs, the `loadScenario` command is executed 13 times when running the full test suite. 
Is this expected?

Reply:
yes, that was happening before this pr as well when calling `loadScenario` directly in every test, the problem with that was that some tests were failing when running isolated and to fix them the only solution was to add a second load scenario in those specific tests (which wasn't very elegant) or add this logic so `loadScenario` is called twice only if needed. Speaking with @arbulu89 we decided that this approach was better so we let every test decide if it needs to load an scenario or not, instead of executing this prior to launch cypress tests as I was initially doing. This way we can still use cypress helper and we are not "hiding" things to the tests by doing them before starting.

Reply:
@vicenteqa
question: why do we need an additional `isTestDataLoaded` command? This command is not used in any tests. The commands are supposed to be "public" functions available for real tests.
Why don't you put the code directly in `preloadTestData`? I find it better.
Otherwise just a simple js function.
I don't think we should be exposing functions that we don't use anywhere else for "code structuring"

Reply:
totally agree about the cy command, I added that function for a better readability and understanding of  what `preloadTestData` command does so I would go and just make a it simple js function.

Reply:
Sorry guys, I still think this code doesn't do what is expected. 

What is needed:
* test data to be loaded 2 times
* test data to be the same, no matter if we run the spec individually, in small batches, or all together (that is: it is always loaded twice)

What we have:
* test data is loaded once if the spec is executed independently
* test data is loaded N times, where N is the number of specs in the batch

Am I wrong?

Reply:
@balanza I think the assumption misses some points.

> What is needed:
>
> -    test data to be loaded 2 times

Only if we don't have any previous data, the database is empty. If we already have data we don't need to load twice

> -   test data to be the same, no matter if we run the spec individually, in small batches, or all together (that is: it is always loaded twice)

We achieve this rerunning the 27 nodes at least once before each execution as this restores changes applied by previous test execution.

> What we have:
>
>  - test data is loaded once if the spec is executed independently

With the current change, the scenario is loaded twice only if we have an empty database, which satisfies our need
>   - test data is loaded N times, where N is the number of specs in the batch

We need this as well, as previous tests might have applied changes that we need to restore running the 27 nodes scenario again (this is what we do right now, before the parallelization)

Reply:
Ok I didn't get it at first, thanks